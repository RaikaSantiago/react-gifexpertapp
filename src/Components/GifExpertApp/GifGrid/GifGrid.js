import React from 'react';
import './GifGrid.css';
import { GifGridItem } from './GifGridItem/GifGridItem';
import { useFechGif } from '../../../hooks/useFechGif';

export const GifGrid = ({ cat }) => {
    const {data:imgs, loading} = useFechGif(cat);

    const getCapitalize = (word) => {
        return word[0].toUpperCase() + word.slice(1).toLowerCase();
    }
    const title = getCapitalize(cat);
    return (
        <>
            <h3 className='animate__animated animate__flash'>{title}</h3>
            {loading && <p>'Cargando...'</p>}
            <div className='card-grid'>
                {
                    imgs.map(img => (
                    //    <GifGridItem key={img.id} img={img}/>
                    <GifGridItem key={img.id} {...img}/>
                    ))
                }
            </div>
        </>
    )
}
