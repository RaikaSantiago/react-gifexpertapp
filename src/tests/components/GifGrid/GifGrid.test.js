import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../../Components/GifExpertApp/GifGrid/GifGrid';

describe('Pruebas <GifGrid  />', () => { 
    const cat = 'Homero';
    let wrapper = shallow( <GifGrid  cat={cat}  />);
    beforeEach( () => {
        wrapper = shallow( <GifGrid cat={cat} />);
    })

    test('Debe de mostrar <GifGrid  /> correctamente', () => { 

        expect( wrapper).toMatchSnapshot();
    })

})