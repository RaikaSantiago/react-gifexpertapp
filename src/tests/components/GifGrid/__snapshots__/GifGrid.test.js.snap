// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifGrid  /> Debe de mostrar <GifGrid  /> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flash"
  >
    Homero
  </h3>
  <p>
    'Cargando...'
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
